### PR TITLE
feat(aom): support new resource alarm rule

### PR DESCRIPTION
**Pull request by saf3dfsa**
**What this PR does / why we need it**:
Support new resource alarm rule.

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST="./huaweicloud/services/acceptance/aom" TESTARGS="-run  TestAccAlarmRuleV4_"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/aom -v -run  TestAccAlarmRuleV4_ -timeout 360m -parallel 4
=== RUN   TestAccAlarmRuleV4_basic
=== PAUSE TestAccAlarmRuleV4_basic
=== RUN   TestAccAlarmRuleV4_metric
=== PAUSE TestAccAlarmRuleV4_metric
=== CONT  TestAccAlarmRuleV4_basic
=== CONT  TestAccAlarmRuleV4_metric
--- PASS: TestAccAlarmRuleV4_metric (58.17s)
--- PASS: TestAccAlarmRuleV4_basic (66.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/aom       66.358s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image

### CODE REVIEW

File: docs/resources/aom_alarm_rule_v4.md [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+    bind_notification_rule_id = var.action_rule_id
+  }
+
+  event_alarm_spec {
+    event_source = "CCE"
+    alarm_source = "systemEvent"
+
+    monitor_objects = [
+      {
+        event_name = "扩容节点超时##ScaleUpTimedOut;数据卷扩容失败##VolumeResizeFailed"
```
Comment (Jason-Zhang9309) [inline docs/resources/aom_alarm_rule_v4.md]: does the enent_name must contain Chinese? How to use it if the customer does know Chinese?

File: docs/resources/aom_alarm_rule_v4.md [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+
+* `alarm_notifications` - (Required, List) Specifies the alarm notification module.
+  The [alarm_notifications](#block--alarm_notifications) structure is documented below.
+
+* `description` - (Optional, String) Specifies the alarm rule description.
+
+* `enable` - (Optional, Bool) Specifies whether to enable the alarm rule. Defaults to **false**.
+
+* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID to which the alarm rule belongs.
+  Changing this creates a new resource.
```
Comment (Jason-Zhang9309) [inline docs/resources/aom_alarm_rule_v4.md]: is this updatable？

File: docs/resources/aom_alarm_rule_v4.md (hunk 1 above)
Comment (saf3dfsa) [reply docs/resources/aom_alarm_rule_v4.md]: It's required.

File: docs/resources/aom_alarm_rule_v4.md [hunk 3]
```diff
@@ -0,0 +1,424 @@
+---
+subcategory: "Application Operations Management (AOM)"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_aom_alarm_rule_v4"
```
Comment (houpeng80) [inline docs/resources/aom_alarm_rule_v4.md]:
```suggestion
page_title: "HuaweiCloud: huaweicloud_aomv4_alarm_rule"
```
fix the format like the resource of huaweicloud_dms_kafkav2_smart_connect_task

File: docs/resources/aom_alarm_rule_v4.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+
+* `promql` - (Required, String) Specifies the prometheus statement.
+
+* `aggregate_type` - (Optional, String) Specifies the Aggregation mode.
+  Valid values are **by**, **avg**, **max**, **min** and **sum**.
+
+* `aggregation_type` - (Optional, String) Specifies the Statistical mode.
+  Valid values are **average**, **minimum**, **maximum**, **sum** and **sampleCount**.
+
+* `aggregation_window` - (Optional, String) Specifies the Statistical period.
```
Comment (houpeng80) [inline docs/resources/aom_alarm_rule_v4.md]:
```suggestion
* `aggregation_window` - (Optional, String) Specifies the statistical period.
```

File: docs/resources/aom_alarm_rule_v4.md [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+
+* `notify_resolved` - (Optional, Bool) Specifies whether to send a notification when an alarm is cleared.
+  Defaults to **false**.
+
+* `notify_triggered` - (Optional, Bool) Specifies whether to send a notification when an alarm is triggered.
+  Defaults to **false**.
+
+* `notify_frequency` - (Optional, String) Specifies the notification frequency.
+  If the **notification_type** is **alarm_policy**, set this parameter to **-1**.
+  If the **notification_type** is **direct**, set this parameter to any of the following:
```
Comment (houpeng80) [inline docs/resources/aom_alarm_rule_v4.md]:
```suggestion
  If the `notification_type` is **alarm_policy**, set this parameter to **-1**.
  If the `notification_type` is **direct**, set this parameter to any of the following:
```

**Pull request merged**